frontend/src/components/CreateModal.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import { Modal, Input, InputNumber, Button, Checkbox, Table, Tag } from 'antd';
+import _ from 'lodash';
+import axios from 'axios';
+
+import { api_endpoints } from '../helpers/constants'
+
+
+class CreateModal extends Component {
+  state = {
+    office_name:'',
+    street: '',
+    city: '',
+    postal_code: '',
+    monthly_rent: '',
+    company_name: '',
+    headquarter: false,
+    offices: []
+  }
+
+  handleAdd = () => {
+    const { office_name, street, city, postal_code, monthly_rent, headquarter } = this.state
+    this.setState({
+      offices: [...this.state.offices, {
+        key: (this.state.offices.length + 1).toString(),
+        name: office_name,
+        street,
+        city,
+        postal_code,
+        monthly_rent,
+        headquarter
+      }]
+    },()=>{
+      console.log(this.state.offices)
+
+      this.setState({
+        office_name: '',
+        street: '',
+        city: '',
+        postal_code: '',
+        monthly_rent: '',
+        headquarter: false
+      })
+    })
+  }
+
+  columns = [
+    {
+      title: 'Office Name',
+      dataIndex: 'name',
+      key: 'name',
+      render: text => <span>{text}</span>
+    },
+    {
+      title: 'Street',
+      dataIndex: 'street',
+      key: 'street',
+      render: text => <span>{text}</span>
+    },
+    {
+      title: 'Other',
+      dataIndex: 'tag',
+      key: 'headquarter',
+      render: (value, record) => {
+        if (record.headquarter) {
+          return (
+            <Tag color="green" >
+              Headquarter
+            </Tag>
+          )
+        }
+
+      },
+    }
+  ]
+
+  handleCancel = () => {
+    this.setState({
+      offices: []
+    }, () => {
+      this.props.onCancel()
+    })
+  }
+
+  handleOk = async () => {
+    const { company_name, offices } = this.state
+    const { onSave } = this.props
+    await axios.post(api_endpoints.CREATE_COMPANY_OFFICE, {
```

Comment:
Bad here, need to define action out side of component, you can just do it some other where like action directory.

Reply:
Actually, we don't need to use redux action here since we don't want to save or store some results from the backend in this api call. But I agree with your idea. I already defined all api services inside `helps` directory. So I could define it there, too. My fault.